### PR TITLE
test(multicluster): add initial telemetry validation test

### PR DESCRIPTION
**Pull request by douglas-reid**
Adds initial telemetry tests to the multicluster scenarios.

This PR is the starting point for validation of multicluster telemetry. At this point in time, it simply attempts to validate that `source_cluster` and `destination_cluster` labels are available in the telemetry. It does not attempt to validate that they have any specific value or similar. These tests will hopefully be expanded upon in the 1.7 release cycle.

[ X ] Networking
[ X ] Policies and Telemetry
[ X ] Test and Release

### CODE REVIEW

**Comment by douglas-reid** [comment]
@howardjohn my current thinking is that if you want to do any advanced querying or the like, I think using prom makes sense (no reason to rewrite prom in a test). probably also makes sense if you are testing for something that involves wider validation, including validating prom functionality itself.

here, i just wanted to start simple and check labels at the source/destination without over-complicating the tests. once I get it working, and we start to improve these tests, maybe it will make sense to add some prom querying??

**Comment by howardjohn** [comment]
I have thought about it a bit before. One benefit is its more direct, without needing to go through scraping intervals, etc so its probably a bit quicker. We could also utilize the clear stats endpoint which would be useful for negative testing for example. Obvious downside is we aren't fully testing prometheus and re-implementing parts of prom. I think it makes sense here, especially since I don't think we have a prometheus config that works with all of these multi cluster scenarios (or do we? If we do we probably do want to use it as a means of testing it?)

**Comment by douglas-reid** [comment]
> I think it makes sense here, especially since I don't think we have a prometheus config that works with all of these multi cluster scenarios (or do we? If we do we probably do want to use it as a means of testing it?)

I don't think we have a good one, and tailoring one that makes easier cross-cluster validation will probably take some work.

We should probably work to add one and expand the existing reachability tests, etc., to check prom to confirm telemetry matches.

**Comment by mandarjog** [comment]
Direct scraping is fine for this test.
We do not need retest all stats testing that we do within a cluster. We are primarily testing that metadata exchange happens correctly. Stats plugin proper is not aware of clusters beyond how it affects labels.

@douglas-reid if we come across specific tests that require full Prometheus, we should add it. Until then this is great.

The types of tests I am thinking about are complex tests, say locality based load balancing test that verifies behaviour using stats. Those will need full prom.

**Comment by douglas-reid** [comment]
Gentle ping for review here.

File: pkg/test/framework/components/echo/docker/sidecar.go [hunk 1]
```diff
@@ -173,6 +188,22 @@ func (s *sidecar) adminRequest(path string, out proto.Message) error {
 	return nil
 }
 
+func (s *sidecar) proxyStats() (map[string]*dto.MetricFamily, error) {
+	arg := fmt.Sprintf("http://%s:%d/stats/prometheus", localhost, proxyAdminPort)
+	result, err := s.container.Exec(context.Background(), "curl", arg)
```
Comment (howardjohn) [inline pkg/test/framework/components/echo/docker/sidecar.go:193]: nit: this will fail on distroless. Any reason you cannot use adminRequest?

Comment (howardjohn) [reply pkg/test/framework/components/echo/docker/sidecar.go:193]: oh nevermind this is native. Using adminRequest still might apply though?

**Pull request merged**